admob/src_java/com/google/firebase/admob/internal/cpp/RewardedAdHelper.java

```diff
@@ -0,0 +1,371 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.admob.internal.cpp;
+
+import android.app.Activity;
+
+import com.google.android.gms.ads.AdError;
+import com.google.android.gms.ads.AdRequest;
+import com.google.android.gms.ads.AdValue;
+import com.google.android.gms.ads.FullScreenContentCallback;
+import com.google.android.gms.ads.LoadAdError;
+import com.google.android.gms.ads.OnPaidEventListener;
+import com.google.android.gms.ads.OnUserEarnedRewardListener;
+import com.google.android.gms.ads.rewarded.ServerSideVerificationOptions;
+import com.google.android.gms.ads.rewarded.RewardedAd;
+import com.google.android.gms.ads.rewarded.RewardedAdLoadCallback;
+import com.google.android.gms.ads.rewarded.RewardItem;
+
+import android.util.Log;
+
+/**
+ * Helper class to make interactions between the AdMob C++ wrapper and Java {@link RewardedAd}
+ * objects cleaner. It's designed to wrap and adapt a single instance of {@link RewardedAd},
+ * translate calls coming from C++ into their (typically more complicated) Java equivalents.
+ */
+public class RewardedAdHelper {
+
+  // C++ nullptr for use with the callbacks.
+  private static final long CPP_NULLPTR = 0;
+
+  // Pointer to the RewardedAdInternalAndroid object that created this
+  // object.
+  private long mRewardedAdInternalPtr;
+
+  // The GMA SDK {@link RewardedAd} associated with this helper.
+  private RewardedAd mRewarded;
+
+  // Synchronization object for thread safe access to:
+  // * mRewarded
```

Comment:
It doesn't actually seem to be locking when mRewarded is used, is this because mRewarded is only ever used in the UI thread? If so you can mention that

Reply:
Adding it for other operations...

Reply:
Done.